src/kernels/jupyter/serverSelector.ts

```diff
@@ -663,3 +676,288 @@ class JupyterServerSelector_Original implements IJupyterServerSelector {
+                if (url && e.button.tooltip === DataScience.removeRemoteJupyterServerEntryInQuickPick()) {
+                    pendingUpdatesToUri = pendingUpdatesToUri.then(() =>
+                        this.serverUriStorage.removeUri(url).catch((ex) => traceError('Failed to update Uri list', ex))
+                    );
+                    items.splice(items.indexOf(e.item), 1);
+                    onDidChangeItems.fire(items.concat([]));
+                }
+            },
+            onDidChangeItems: onDidChangeItems.event
+        });
+        await pendingUpdatesToUri.catch((ex) => traceError('Failed to update Uri list', ex));
+        if (item.label === this.localLabel) {
+            await this.setJupyterURIToLocal();
+        } else if (!item.newChoice && !item.provider) {
+            await this.setJupyterURIToRemote(!isNil(item.url) ? item.url : item.label, false, item.label);
+        } else if (!item.provider) {
+            return this.selectRemoteURI.bind(this);
+        } else {
+            return this.selectProviderURI.bind(this, item.provider, item);
+        }
+    }
+
+    private async selectProviderURI(
+        provider: IJupyterUriProvider,
+        item: ISelectUriQuickPickItem,
+        _input: IMultiStepInput<{}>,
+        _state: {}
+    ): Promise<InputStep<{}> | void> {
+        if (!provider.handleQuickPick) {
+            return;
+        }
+        const result = await provider.handleQuickPick(item, true);
+        if (result === 'back') {
+            throw InputFlowAction.back;
+        }
+        if (result) {
+            await this.handleProviderQuickPick(provider.id, result);
+        }
+    }
+    private async handleProviderQuickPick(id: string, result: JupyterServerUriHandle | undefined) {
+        if (result) {
+            const uri = generateUriFromRemoteProvider(id, result);
+            await this.setJupyterURIToRemote(uri);
+        }
+    }
+    private async selectRemoteURI(input: IMultiStepInput<{}>, _state: {}): Promise<InputStep<{}> | void> {
+        let initialValue = defaultUri;
+        try {
+            const text = await this.clipboard.readText().catch(() => '');
+            const parsedUri = Uri.parse(text.trim(), true);
+            // Only display http/https uris.
+            initialValue = text && parsedUri && parsedUri.scheme.toLowerCase().startsWith('http') ? text : defaultUri;
+        } catch {
+            // We can ignore errors.
+        }
+        // Ask the user to enter a URI to connect to.
+        const uri = await input.showInputBox({
+            title: DataScience.jupyterSelectURIPrompt(),
+            value: initialValue || defaultUri,
+            validate: this.validateSelectJupyterURI,
+            prompt: ''
+        });
+
+        // Offer the user a change to pick a display name for the server
+        // Prepopulate the value with the uri as the default suggestiong
+        const newDisplayName = await this.applicationShell.showInputBox({
```

Comment:
This is the only big change from the previous server picker, new input box to select a display name. Pre-populated with the URL, so you can just hit enter again to get the previous UI.